### PR TITLE
CppCheck HWBaseboardRadiator

### PR DESCRIPTION
**Pull request by rraustad**
Pull request overview
---------------------
 - Fixes CppCheck issues
 - DESCRIBE PURPOSE OF THIS PULL REQUEST

**NOTE: ENHANCEMENTS MUST FOLLOW A SUBMISSION PROCESS INCLUDING A FEATURE PROPOSAL AND DESIGN DOCUMENT PRIOR TO SUBMITTING CODE**

### Pull Request Author
Add to this list or remove from it as applicable.  This is a simple templated set of guidelines.
 - [ ] Title of PR should be user-synopsis style (clearly understandable in a standalone changelog context)
 - [ ] Label the PR with at least one of: Defect, Refactoring, NewFeature, Performance, and/or DoNoPublish
 - [ ] Pull requests that impact EnergyPlus code must also include unit tests to cover enhancement or defect repair
 - [ ] Author should provide a "walkthrough" of relevant code changes using a GitHub code review comment process
 - [ ] If any diffs are expected, author must demonstrate they are justified using plots and descriptions
 - [ ] If changes fix a defect, the fix should be demonstrated in plots and descriptions
 - [ ] If any defect files are updated to a more recent version, upload new versions here or on DevSupport
 - [ ] If IDD requires transition, transition source, rules, ExpandObjects, and IDFs must be updated, and add IDDChange label
 - [ ] If structural output changes, add to output rules file and add OutputChange label
 - [ ] If adding/removing any LaTeX docs or figures, update that document's CMakeLists file dependencies

### Reviewer
This will not be exhaustively relevant to every PR.
 - [ ] Perform a Code Review on GitHub
 - [ ] If branch is behind develop, merge develop and build locally to check for side effects of the merge
 - [ ] If defect, verify by running develop branch and reproducing defect, then running PR and reproducing fix
 - [ ] If feature, test running new feature, try creative ways to break it
 - [ ] CI status: all green or justified
 - [ ] Check that performance is not impacted (CI Linux results include performance check)
 - [ ] Run Unit Test(s) locally
 - [ ] Check any new function arguments for performance impacts
 - [ ] Verify IDF naming conventions and styles, memos and notes and defaults
 - [ ] If new idf included, locally check the err file and other outputs

### CODE REVIEW

File: src/EnergyPlus/HWBaseboardRadiator.cc [hunk 1]
```diff
@@ -1440,39 +1318,34 @@ namespace HWBaseboardRadiator {
...
-        WaterMassFlowRate = state.dataLoopNodes->Node(HWBaseboard(BaseboardNum).WaterInletNode).MassFlowRate;
+        auto &hWBaseboard = state.dataHWBaseboardRad->HWBaseboard(BaseboardNum);
+
+        int const ZoneNum = hWBaseboard.ZonePtr;
+        auto const &zeroSourceSumHATsurf = state.dataHWBaseboardRad->ZeroSourceSumHATsurf(ZoneNum);
+        Real64 QZnReq = state.dataZoneEnergyDemand->ZoneSysEnergyDemand(ZoneNum).RemainingOutputReqToHeatSP;
+        Real64 AirInletTemp = hWBaseboard.AirInletTemp;
+        Real64 WaterInletTemp = hWBaseboard.WaterInletTemp;
+        Real64 WaterMassFlowRate = state.dataLoopNodes->Node(hWBaseboard.WaterInletNode).MassFlowRate;
         HWBaseboardDesignData HWBaseboardDesignDataObject{
```
Comment (amirroth) [inline src/EnergyPlus/HWBaseboardRadiator.cc]: You are creating a local copy of this object. Is that the intent? If not, change to `const &`.

File: src/EnergyPlus/HWBaseboardRadiator.cc [hunk 2]
```diff
@@ -1557,41 +1430,20 @@ namespace HWBaseboardRadiator {
...
         int WaterInletNode;
         int WaterOutletNode;
-        auto &HWBaseboard = state.dataHWBaseboardRad->HWBaseboard;
-        auto &LastSysTimeElapsed = state.dataHWBaseboardRad->LastSysTimeElapsed;
-        auto &QBBRadSrcAvg = state.dataHWBaseboardRad->QBBRadSrcAvg;
-        auto &LastQBBRadSrc = state.dataHWBaseboardRad->LastQBBRadSrc;
-        auto &QBBRadSource = state.dataHWBaseboardRad->QBBRadSource;
-        auto &LastTimeStepSys = state.dataHWBaseboardRad->LastTimeStepSys;
+        auto const &hWBaseboard = state.dataHWBaseboardRad->HWBaseboard(BaseboardNum);
+        auto &lastSysTimeElapsed = state.dataHWBaseboardRad->LastSysTimeElapsed(BaseboardNum);
```
Comment (amirroth) [inline src/EnergyPlus/HWBaseboardRadiator.cc]: All of these are bad uses of local references. If the variable is read only, then copy by value. If the value is written, don't use a reference at all. Also, they should not be `auto`.

File: src/EnergyPlus/HWBaseboardRadiator.cc [hunk 3]
```diff
@@ -1704,49 +1533,43 @@ namespace HWBaseboardRadiator {
...
 
         // Initialize arrays
         state.dataHeatBalFanSys->SurfQHWBaseboard = 0.0;
         state.dataHeatBalFanSys->ZoneQHWBaseboardToPerson = 0.0;
 
         for (BaseboardNum = 1; BaseboardNum <= NumHWBaseboards; ++BaseboardNum) {
+            auto &HWBaseboard = state.dataHWBaseboardRad->HWBaseboard(BaseboardNum);
+            auto const &QBBRadSource = state.dataHWBaseboardRad->QBBRadSource(BaseboardNum);
 
             HWBaseboardDesignData HWBaseboardDesignDataObject{
```
Comment (amirroth) [inline src/EnergyPlus/HWBaseboardRadiator.cc]: Making a local copy of the design object again.

File: src/EnergyPlus/HWBaseboardRadiator.cc [hunk 4]
```diff
@@ -1855,26 +1670,26 @@ namespace HWBaseboardRadiator {
...
-                                    HWBaseboard(BaseboardNum).WaterMassFlowRate);
-
-        PullCompInterconnectTrigger(state,
-                                    HWBaseboard(BaseboardNum).plantLoc,
-                                    HWBaseboard(BaseboardNum).BBInletTempFlowReSimIndex,
-                                    HWBaseboard(BaseboardNum).plantLoc,
-                                    DataPlant::CriteriaType::Temperature,
-                                    HWBaseboard(BaseboardNum).WaterOutletTemp);
+        PlantUtilities::PullCompInterconnectTrigger(state,
+                                                    HWBaseboard(BaseboardNum).plantLoc,
```
Comment (amirroth) [inline src/EnergyPlus/HWBaseboardRadiator.cc:1635]: `auto &hwBaseboard`

File: src/EnergyPlus/HWBaseboardRadiator.cc (hunk 2 above)
Comment (rraustad) [reply src/EnergyPlus/HWBaseboardRadiator.cc]: I'm confused. These were changed to be references to objects. How could these not be `auto`? I can certainly replace `lastSysTimeElapsed` with `state.dataHWBaseboardRad->LastSysTimeElapsed(BaseboardNum)` but I don't see what's wrong with what is here now.

Comment (amirroth) [reply src/EnergyPlus/HWBaseboardRadiator.cc]: The array `state.dataHWBaseboardRad->LastSysTimeElapsed` is not an array of objects, it's an array of `Real64`'s. And so a reference to any element in it is a reference to a `Real64`, not to an object.

Comment (mjwitte) [reply src/EnergyPlus/HWBaseboardRadiator.cc]: Different question, why is `LastSysTimeElapsed` a separate array and not part of `HWBaseboard`?

Comment (amirroth) [reply src/EnergyPlus/HWBaseboardRadiator.cc]: You can ask that question about 50% of the arrays in EnergyPlus.

**Comment by rraustad** [comment]
@Myoldmopar this is the branch that fails to run the unit tests locally on Windows and not surprisingly fails on the HWBaseboard unit tests. The unit tests don't like something I did here but code inspection does not show anything that stands out. These unit tests run in Debug and RelWithDebug (i.e., exits with code 0) yet fails in Release without any indication of why. I put print statements in the unit test and the unit test runs to completion and then something happens during unit test clean up that fails. I am not sure how to diagnose this more that I have done so far.

image

**Comment by rraustad** [comment]
Example of diffs. These are small but something is causing this. 
[5ZoneCoolingPanelBaseboardVarOff.zip](https://github.com/NREL/EnergyPlus/files/11493335/5ZoneCoolingPanelBaseboardVarOff.zip)

**Comment by rraustad** [comment]
I am not wasting any more time on this branch. I will delete this and start over.

**Comment by rraustad** [comment]
@Myoldmopar I ran Git bisect locally and it tells me this commit has no diffs [05bcdff ref thisHWBaseboardDesign](https://github.com/NREL/EnergyPlus/pull/10019/commits/05bcdffefb45d458a683371695684161fba7e85b). I used Git reset locally and pushed the change. The problem is when I committed locally this morning there were no changed files to commit so I pushed anyway. That last commit above (5df4d26 Revert part of accounting) is a change I made last night to see if I could find the diff issue. So I don't think I actually reverted this branch to the clean state of 05bcdff.

**Comment by rraustad** [comment]
replaced by #10023.

**Pull request closed without merging**